src/game/client/gameclient.cpp

```diff
@@ -409,9 +419,177 @@ void GameClient::Reset()
+
+                        g_theWindowManager->Update();
+                        g_theDisplay->Draw();
+                        rts::Sleep_Ms(100);
+                    }
+
+                    Set_FP_Mode();
+                    layout->Destroy_Windows();
+                    layout->Delete_Instance();
+                }
+
+                g_theWriteableGlobalData->m_unkBool26 = true;
+            }
+
+            g_theShell->Show_Shell_Map(1);
+            g_theShell->Show_Shell(1);
+            g_theWriteableGlobalData->m_afterIntro = false;
+        }
+    }
+
+    if (g_theSnowManager != nullptr) {
+        g_theSnowManager->Update();
+    }
+
+    g_theAnim2DCollection->Update();
+
+    if (g_theKeyboard != nullptr) {
+        g_theKeyboard->Update();
+        g_theKeyboard->Create_Stream_Messages();
+    }
+
+    g_theEva->Update();
+
+    if (g_theMouse != nullptr) {
+        g_theMouse->Update();
+        g_theMouse->Create_Stream_Messages();
+    }
+
+    if (g_theInGameUI->Is_Camera_Tracking_Drawable()) {
+        Drawable *drawable = g_theInGameUI->Get_First_Selected_Drawable();
+
+        if (drawable != nullptr) {
+            g_theTacticalView->Look_At(drawable->Get_Position());
+        } else {
+            g_theInGameUI->Set_Camera_Drawing_Trackable(false);
+        }
+    }
+
+    if (g_theWriteableGlobalData->m_playIntro || g_theWriteableGlobalData->m_afterIntro) {
+        g_theDisplay->Draw();
+        g_theDisplay->Update();
+    } else {
+        g_theWindowManager->Update();
+        g_theVideoPlayer->Update();
+
+        bool b = g_theTacticalView->Is_Time_Frozen() && !g_theTacticalView->Is_Camera_Movement_Finished();
+        b = b || g_theScriptEngine->Is_Time_Frozen_Debug();
+        b = b || g_theScriptEngine->Is_Time_Frozen_Script();
+        b = b || g_theGameLogic->Is_Game_Paused();
+        unsigned int index;
+
+        if (g_thePlayerList != nullptr) {
+            index = g_thePlayerList->Get_Local_Player()->Get_Player_Index();
+        } else {
+            index = 0;
+        }
+
+        static unsigned int lastFrame = -1;
+
+        b = b || lastFrame == m_frame;
+        lastFrame = m_frame;
+
+        if (!b) {
+#ifdef GAME_DEBUG_STRUCTS
+            if (g_theWriteableGlobalData->m_shroudOn)
 #endif
+            {
+                g_theGhostObjectManager->Update_Orphaned_Objects(nullptr, 0);
+            }
+
+            Drawable *next_drawable;
+
+            for (Drawable *draw = First_Drawable(); draw != nullptr; draw = next_drawable) {
+                next_drawable = draw->Get_Next();
+#ifdef GAME_DEBUG_STRUCTS
+                if (g_theWriteableGlobalData->m_shroudOn)
+#endif
+                {
+                    Object *obj = draw->Get_Object();
+
+                    if (obj != nullptr) {
+                        ObjectShroudStatus status = obj->Get_Shrouded_Status(index);
+
+                        if (status >= SHROUDED_UNK3 && draw->Get_Remain_Visible_Frames() != 0) {
```

Comment:
SHROUDED_PARTIAL = SHROUDED_TRANSITION // Object has entered fog and will be hidden soon
SHROUDED_UNK3 = SHROUDED_SEEN // Object is hidden in fog and has been seen before
SHROUDED_UNK4 = SHROUDED_NEVERSEEN // Object is hidden in fog and has never been seen before

Reply:
fixed.